sycl/include/CL/sycl/handler.hpp

```diff
@@ -366,6 +367,107 @@ class __SYCL_EXPORT handler {
     return true;
   }
 
+  /// Handles some special cases of the copy operation from one accessor
+  /// to another accessor. Returns true if the copy is handled here.
+  ///
+  /// \param Src is a source SYCL accessor.
+  /// \param Dst is a destination SYCL accessor.
+  // TODO: support atomic accessor in Src or/and Dst.
+  template<
+      typename TSrc, int DimSrc, access::mode ModeSrc, access::target TargetSrc,
+      typename TDst, int DimDst, access::mode ModeDst, access::target TargetDst,
+      access::placeholder IsPHSrc, access::placeholder IsPHDst>
+  detail::enable_if_t<(DimSrc > 0) && (DimDst > 0), bool>
```

Comment:
Should we check here if `DimSrc == DimDst`?

Reply:
I don't know if it is correct to copy from 1D accessor to 3D. It is probably correct. So, that check is not needed.
In my patch I do not change any constraints. 
My patch only separates the existing code to this routine to make it possible to copy to/from 0-dim accessor.